locallib.php

```diff
@@ -379,11 +379,16 @@ private function process_action_edit_choice() {
 
                 if (!$mform->is_cancelled()) {
                     if ($mform->is_validated()) {
+                        if($data->usegroups) {
+                            $this->update_choice_groups($data->choiceid, $data->groupselector);
```

Comment:
> For new choices there is no choiceid when called for the first time. Should check if($data->usegroups && !empty($data->choiceid))

passing the $choiceid allows the code to get existing groups associated with the choice - if no choiceid is passed (it's empty) it will end up seeing there are no groups for this choice. - but I agree - that code needs to be clearer/safer.

Reply:
hmm - now I look closer at that code I'm not so sure... been a while since  I've looked at this...

Reply:
yeah - was just in the wrong place - with all the other changes we had going on that part got moved to the wrong part of that function - I've moved it further down (after the choice save) and that's fixed it.